frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/RediscoverSuggester.java

```diff
@@ -71,4 +80,32 @@ private long daysDiffTillNow(long lastModifiedTimeMillis) {
     val millisDiff = currentTimeMillis - lastModifiedTimeMillis;
     return millisDiff / (24 * 60 * 60 * 1000);
   }
+
+  public void enqueueChecksThenMaybePerform() {
+    new Task.Backgroundable(
+        gitRepository.getProject(),
+        getString("string.GitMachete.RediscoverSuggester.backgroundable-check-task.title")) {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        if (!areAllLocalBranchesManaged()) {
+          ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> perform());
+        }
+      }
+    }.queue();
+  }
+
+  private boolean areAllLocalBranchesManaged() {
+    val localBranches = gitRepository.getBranches().getLocalBranches();
+    val branchLayoutReader = RuntimeBinding.instantiateSoleImplementingClass(IBranchLayoutReader.class);
+    try {
+      val branchLayout = branchLayoutReader.read(gitRepository.getMacheteFilePath());
+      val areLocalBranchesManagedSeq = localBranches.stream().map(GitReference::getName)
+          .map(branchLayout::findEntryByName)
+          .collect(Collectors.toList());
+
+      return Option.sequence(areLocalBranchesManagedSeq).isDefined();
```

Comment:
nice 👍

Reply:
Hmmm are you checking here whether **all** local branches are managed, or whether **any** local branch is managed? 🤔

Reply:
`Option.sequence()` will be `None` if any element in the sequence are `None`.

Reply:
> Option.sequence() will be None if any element in the sequence are None.

Whoops that's very non-obvious... I wouldn't rely on that, pls make the check more explicit.

In fact I'm now not even sure if [Cats](https://typelevel.org/cats/) which is probably the source of inspiration for Vavr wrt. `sequence`/`traverse` would behave the same way (probably it would, but again, I wouldn't rely on that in code that's supposed to future-proof/maintainable)